CustomControlLibrary/NumberToStringConvertor.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using System.Windows.Data;
+
+namespace CustomControlLibrary
+{
+    /// <summary>
+    /// Converts string in and from of double type.
+    /// </summary>
+    public class NumberToStringConvertor : IValueConverter
+    {
+        /// <inheritdoc/>
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            var result = (double)value;
+            return result.ToString(CultureInfo.CurrentUICulture.NumberFormat);
+        }
+
+        /// <inheritdoc/>
+        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            var processText = ProcessTextToNumberRepresentation(value as string);
+
+            if (!double.TryParse(processText, NumberStyles.Number, CultureInfo.CurrentUICulture, out double numberResult)
+                || processText.IndexOf(CultureInfo.CurrentUICulture.NumberFormat.CurrencyDecimalSeparator) == processText.Length - 1)
+            {
+                return Binding.DoNothing;
+            }
+
+            return numberResult;
+        }
+
+        private static string ProcessTextToNumberRepresentation(string input)
```

Comment:
Instead of validating the input text, I think you should handle the OnKeyDown event of textbox to prevent users from enter non-numerical text.
If a control is expected to accept only numeric value, prevent entering non-numerical text is a more common behavior

Reply:
Ok.